qiskit/providers/backend.py

```diff
@@ -229,3 +239,332 @@ class can handle either situation.
             Job: The job object for the run
         """
         pass
+
+
+class BackendV2(Backend, ABC):
+    """Abstract class for Backends
+
+    This abstract class is to be used for all Backend objects created by a
+    provider. This version differs from earlier abstract Backend classes in
+    that the configuration attribute no longer exists instead attributes
+    exposing equivalent required immutable properties of the backend device
+    are added. For example ``backend.configuration().n_qubits`` is accessible
+    from ``backend.num_qubits`` now.
+
+    The ``options`` attribute of the backend is used to contain the dynamic
+    user configurable options of the backend. It should be used more for
+    runtime options that configure how the backend is used. For example,
+    something like a ``shots`` field for a backend that runs experiments which
+    would contain an int for how many shots to execute.
+
+    If migrating a provider from :class:`~qiskit.providers.BackendV1` or
+    :class:`~qiskit.providers.BaseBackend` one thing to keep in mind is for
+    backwards compatibility you might need to add a configuration method that
+    will build a :class:`~qiskit.providers.models.BackendConfiguration` object
+    and :class:`~qiskit.providers.models.BackendProperties` from the attributes
+    defined in this class for backwards compatibility.
+    """
+
+    version = 2
+
+    def __init__(self, provider, **fields):
+        self._options = self._default_options()
+        self._provider = provider
+        self._coupling_map = None
+        if fields:
+            for field in fields:
+                if field not in self._options.data:
+                    raise AttributeError("Options field %s is not valid for this backend" % field)
+            self._options.update_config(**fields)
+        self._basis_gates_all = None
+
+    @property
+    def instructions(self) -> List[Gate]:
+        """A list of :class:`~qiskit.circuit.Instruction` instances that the backend supports."""
+        return list(self.target.instructions)
```

Comment:
Sure I can add a method for that, although nothing in the target works with classical bits but I think returning the list of the combinations makes sense. I'm guessing you would prefer I call that method `instructions` and this one `operations`?

Reply:
Yeah operations for this and instructions for the combination of (operation, qubit(s)) would be great.

I agree clbits can be dropped. For things like measurements the qubit is the one that defines things like duration & error. The clbit can be anything since you can measure into any classical bit you want.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/5885/commits/5c252d46be0f2ad0a9800ba8392e1636ed519ef1